### PR TITLE
Add silo

### PR DESCRIPTION
**Pull request by Aviu00**
## About the PR

Now all lathes are connected together and share the same resources if they are linked to silo.

## Why / Balance

Simplifies resource management.

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl: Aviu
- add: Added the materials silo, a machine that connects all lathes on the station together so that they share resources. You can find it in the comms room on every map.

### CODE REVIEW

**Comment by BombasterDS** [comment]
How did you fix lathe ui without lathe prediction?? 😭

**Comment by Aviu00** [comment]
> How did you fix lathe ui without lathe prediction?? 😭

What

**Comment by BombasterDS** [comment]
> > How did you fix lathe ui without lathe prediction?? 😭
> 
> What

When i tried to make material silo, i had problems with delayed material update in UI

**Comment by Aviu00** [comment]
> Try to test it with both proto and autolathe UIs opened.

It works fine, no delay.

**Comment by deltanedas** [comment]
30 goobux being sent when merged and i get on :trollface:

**Comment by Aidenkrz** [comment]
I think I prefer @BombasterDS solution which required linking instead of just being shared across everything on the same grid.

**Comment by Lyacs** [comment]
kinda feel way too op. And allows resources teleportation between lathes if i understood correctly. Personnaly against it.

**Comment by BombasterDS** [comment]
> kinda feel way too op. And allows resources teleportation between lathes if i understood correctly. Personnaly against it.

This is how it works in ss13 and it's very user-friendly for everyone on station.

**Comment by BombasterDS** [comment]
> I think I prefer @BombasterDS solution which required linking instead of just being shared across everything on the same grid.

@Aviu00 https://github.com/BombasterDS/Goob-Station/tree/material-silo if you want to look into it. I have different logic, but there are working material silo machine that can be connected to lathes. Check it if you want. Not really sure if I fixed UI delay problem there.

**Comment by Aviu00** [comment]
> I think I prefer @BombasterDS solution which required linking instead of just being shared across everything on the same grid.

I can add something like "connect to silo" button to lathes.

**Comment by Piras314** [comment]
I don't mind mapping it onto all maps if you want it to be its own machine.

**Comment by BombasterDS** [comment]
> > I think I prefer @BombasterDS solution which required linking instead of just being shared across everything on the same grid.
> 
> I can add something like "connect to silo" button to lathes.

Main problem with this that you can steal all station resources as antag without doing anything. You can place lathe somewhere in maints and make everything you want without getting into secured area to connect silo.

**Comment by gluesniffler** [comment]
> > > I think I prefer @BombasterDS solution which required linking instead of just being shared across everything on the same grid.
> > 
> > 
> > I can add something like "connect to silo" button to lathes.
> 
> Main problem with this that you can steal all station resources as antag without doing anything. You can place lathe somewhere in maints and make everything you want without getting into secured area to connect silo.

It works out just fine in 13, and as mentioned in discord, this is probably not a big deal because getting a lathe to fabricate stuff either implies you:
1. Steal it from science, or they give it to you, so someone *at least* knows that theres a loose lathe.
2. Steal it from the tech vault (they SHOULD notice)
3. Steal it from a department (they WILL notice)

Additionally to that, the underlying problem is that
1. Theres no visibility on where the materials are going, or who is using them (you could easily make a view that shows the current queue for all lathes, and who started them based off ID)
2. Ore is too scarce so sharing an entire pool with everyone will lead to infighting and issues, thus we should buff the ore output from salvage (lavaland when)

**Comment by BombasterDS** [comment]
> > > > I think I prefer @BombasterDS solution which required linking instead of just being shared across everything on the same grid.
> > > 
> > > 
> > > I can add something like "connect to silo" button to lathes.
> > 
> > 
> > Main problem with this that you can steal all station resources as antag without doing anything. You can place lathe somewhere in maints and make everything you want without getting into secured area to connect silo.
> 
> It works out just fine in 13, and as mentioned in discord, this is probably not a big deal because getting a lathe to fabricate stuff either implies you:
> 
> 1. Steal it from science, or they give it to you, so someone _at least_ knows that theres a loose lathe.
> 2. Steal it from the tech vault (they SHOULD notice)
> 3. Steal it from a department (they WILL notice)
> 
> Additionally to that, the underlying problem is that
> 
> 1. Theres no visibility on where the materials are going, or who is using them (you could easily make a view that shows the current queue for all lathes, and who started them based off ID)
> 2. Ore is too scarce so sharing an entire pool with everyone will lead to infighting and issues, thus we should buff the ore output from salvage (lavaland when)

In SS13 (TG build) you have to connect lathes to ore silo in vault. They roundstart connected, but you need to get into vault to connect your own lathe. Right know i'm telling about problem when you can connect lathe just by clicking in lathe menu. You can do it anytime, steal as much as you want and be untrackable (in ss13 ore silo have UI list with all connected devices and logs of inserting/ejecting materials and you able to disconnect lathe from there)

**Comment by BombasterDS** [comment]
Still, lack of material on station is big problem. Salvage ofter is useless team that do nothing for station, just flying on their shuttle for fun (now i'm sure that deleting salv shuttle and giving them jetpacks is actually good idea). I really want to see miners in ss14, but i don't really like lavaland idea where you can hide from everyone whenever you want.

**Comment by JoeHammad1844** [comment]
YYYEEEESSS

**Comment by JoeHammad1844** [comment]
by the way what happens if one lathe on one grid has, say, 10 steel and 10 plastic, and gets anchored onto a new grid that has 10 steel and 0 plastic?

**Comment by JustSome1Nearby** [comment]
POV that one silly cargonian synd who emagged the silo and turned all your hard earned mats into ammo

**Comment by JoeHammad1844** [comment]
> POV that one silly cargonian synd who emagged the silo and turned all your hard earned mats into ammo

based

**Comment by Aidenkrz** [comment]
Drafting this, feel free to reopen when relevant issues have been addressed.

**Comment by Aviu00** [comment]
It is done.

File: Content.Shared/_Goobstation/CVars/GoobCvars.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using Robust.Shared.Configuration;
+
+namespace Content.Shared._Goobstation.CVars;
+
+[CVarDefs]
+public sealed class GoobCVars : Robust.Shared.CVars
```
Comment (Aidenkrz) [inline Content.Shared/_Goobstation/CVars/GoobCvars.cs:6]: We shouldve been doing this from the start, with next upstream merge we'll have to move like 20 cvars here :sob:

**Review by Aidenkrz — APPROVED** [review]

**Pull request merged**